modules/qortexRtdProvider.js

```diff
@@ -122,45 +244,119 @@ export function loadScriptTag(config) {
+          logWarn('Contexual record is not yet complete at this time')
+        }
+      })
+      .catch((e) => {
+        const errorStatus = e.message;
+        logWarn('Returned error status code: ' + errorStatus)
+      })
+  }
+}
 /**
  * Helper function to set initial values when they are obtained by init
  * @param {Object} config module config obtained during init
  */
 export function initializeModuleData(config) {
-  const DEFAULT_API_URL = 'https://demand.qortex.ai';
-  const {apiUrl, groupId, bidders} = config.params;
-  requestUrl = `${apiUrl || DEFAULT_API_URL}/api/v1/analyze/${groupId}/prebid`;
-  bidderArray = bidders;
-  impressionIds = new Set();
-  currentSiteContext = null;
+  const {apiUrl, groupId, bidders, enableBidEnrichment} = config.params;
+  const qortexUrlBase = apiUrl || DEFAULT_API_URL;
+  const windowUrl = window.top.location.host;
+  qortexSessionInfo.bidEnrichmentDisabled = enableBidEnrichment !== null ? !enableBidEnrichment : true;
+  qortexSessionInfo.bidderArray = bidders;
+  qortexSessionInfo.impressionIds = new Set();
+  qortexSessionInfo.currentSiteContext = null;
+  qortexSessionInfo.pageAnalysisData = {
+    contextRetrieved: false,
+    contextAdded: {}
+  };
+  qortexSessionInfo.sessionId = generateSessionId();
+  qortexSessionInfo.groupId = groupId;
+  qortexSessionInfo.groupConfigUrl = `${qortexUrlBase}/api/v1/prebid/group/configs/${groupId}/${windowUrl}`;
+  qortexSessionInfo.contextUrl = `${qortexUrlBase}/api/v1/prebid/${groupId}/page/lookup`;
+  qortexSessionInfo.analyticsUrl = generateAnalyticsHostUrl(qortexUrlBase);
+  return qortexSessionInfo;
+}
+
+export function saveContextAdded(reqBids, bidders = null) {
+  const id = reqBids.auctionId;
+  const contextBidders = bidders ?? Array.from(new Set(reqBids.adUnits.flatMap(adunit => adunit.bids.map(bid => bid.bidder))))
+  qortexSessionInfo.pageAnalysisData.contextAdded[id] = contextBidders;
 }
 
 export function setContextData(value) {
-  currentSiteContext = value
+  qortexSessionInfo.currentSiteContext = value
+}
+
+export function setGroupConfigData(value) {
+  qortexSessionInfo.groupConfig = value
+}
+
+function generateSessionId() {
+  const randomInt = window.crypto.getRandomValues(new Uint32Array(1));
+  const currentDateTime = Math.floor(Date.now() / 1000);
+  return 'QX' + randomInt.toString() + 'X' + currentDateTime.toString()
+}
+
+function attachContextAnalytics (data) {
+  let qxData = {};
+  let qxDataAdded = false;
+  if (qortexSessionInfo?.pageAnalysisData?.contextAdded[data.auctionId]) {
+    qxData = qortexSessionInfo.currentSiteContext;
+    qxDataAdded = true;
+  }
+  data.qortexData = qxData;
+  data.qortexDataAdded = qxDataAdded;
+  return data;
+}
+
+function shouldSendAnalytics() {
+  const analyticsPercentage = qortexSessionInfo.groupConfig?.prebidReportingPercentage ?? 0;
+  const randomInt = Math.random().toFixed(5) * 100;
+  return analyticsPercentage > randomInt;
+}
+
+function shouldAllowBidEnrichment() {
```

Comment:
what is thi9s controlling if you've removed the text going over the wire

Reply:
it is only used for determining if we enrich the auction with contextual data about the current page (and videos known to be on the page) from our servers. This still requires a check on the prebid config and within our zone configurations to even attempt to retrieve information known about the current page.